builder/docker.go

```diff
@@ -160,26 +160,60 @@ func (b *Builder) getDockerMounts() ([]string, error) {
 	return reduceDockerMounts(mounts), nil
 }
 
-// RunCommandInContainer will pull the content necessary to build a docker
-// image capable of running the desired command, build that image, and then
-// run the command in that image.
-func (b *Builder) RunCommandInContainer(cmd []string) error {
+func (b *Builder) prepareContainer() (string, error) {
 	format, err := b.getUpstreamFormat(b.UpstreamVer)
 	if err != nil {
-		return err
+		return "", err
 	}
-
 	imageName, err := b.getDockerImageName(format)
 	if err != nil {
-		return errors.Wrap(err, "Unable to get docker image name for format "+format)
+		return "", errors.Wrap(err, "Unable to get docker image name for format "+format)
 	}
-
 	fmt.Println("Updating docker image")
-
 	if err = helpers.RunCommand("docker", "pull", imageName); err != nil {
 		fmt.Printf("WARNING: Unable to pull docker image for format %s. Trying with cached image\n", format)
 	}
 
+	return imageName, nil
+
+}
+
+func (b *Builder) prepareContainerOffline() (string, error) {
+	var format string
+	if err := helpers.RunCommandSilent("docker", "image", "inspect", "clearlinux/mixer:"+b.State.Mix.Format); err != nil {
```

Comment:
Shouldn't you use `getDockerImageName()` here instead of `"clearlinux/mixer:" + b.State.Mix.Format`? That string is configurable, so the user might have changed it. If so, the cached image won't have that name.

Reply:
That's a good point, but the user should never change the cached image name otherwise they'd not be able to do auto format bumps anymore.